overlord/configstate/configcore/corecfg.go

```diff
@@ -62,6 +63,100 @@ func validateBoolFlag(tr config.Conf, flag string) error {
 	return nil
 }
 
+// PlainCoreConfig carries a read-only copy of core config and implements
+// config.ConfGetter interface.
+type PlainCoreConfig map[string]interface{}
+
+// Get implements config.ConfGetter interface.
+func (cfg PlainCoreConfig) Get(snapName, key string, result interface{}) error {
+	if snapName != "core" {
+		return fmt.Errorf("internal error: expected core snap in Get(), %q was requested", snapName)
+	}
+
+	// XXX: should we support nesting with dotted key?
+	val, ok := cfg[key]
+	if !ok {
+		return &config.NoOptionError{SnapName: snapName, Key: key}
+	}
+
+	rv := reflect.ValueOf(result)
+	rv.Elem().Set(reflect.ValueOf(val))
+	return nil
+}
+
+// GetMaybe implements config.ConfGetter interface.
+func (cfg PlainCoreConfig) GetMaybe(instanceName, key string, result interface{}) error {
+	err := cfg.Get(instanceName, key, result)
+	if err != nil && !config.IsNoOption(err) {
+		return err
+	}
+	return nil
+}
+
+// ApplyOptions encapsulates extra options passed to individual core config
+// handlers when configuration is applied to a specific root directory with
+// Apply().
+type ApplyOptions struct {
```

Comment:
this strictly doesn't need to be exported,  I think the sticking point is ExportExperimentalFlags but see my comment there

Reply:
let's rename Apply to FilesystemOnlyApply

and this one to fsOnlyContext